pandas/tests/groupby/test_grouping.py

```diff
@@ -814,19 +820,26 @@ def test_groupby_empty(self):
         tm.assert_series_equal(result, expected)
 
         # check group properties
-        assert len(gr.grouper.groupings) == 1
-        tm.assert_numpy_array_equal(
-            gr.grouper.group_info[0], np.array([], dtype=np.dtype(np.intp))
-        )
+        msg = "SeriesGroupBy.grouper is deprecated"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            assert len(gr.grouper.groupings) == 1
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            tm.assert_numpy_array_equal(
+                gr.grouper.group_info[0], np.array([], dtype=np.dtype(np.intp))
+            )
 
-        tm.assert_numpy_array_equal(
-            gr.grouper.group_info[1], np.array([], dtype=np.dtype(np.intp))
-        )
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            tm.assert_numpy_array_equal(
+                gr.grouper.group_info[1], np.array([], dtype=np.dtype(np.intp))
+            )
 
-        assert gr.grouper.group_info[2] == 0
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            assert gr.grouper.group_info[2] == 0
 
         # check name
-        assert s.groupby(s).grouper.names == ["name"]
+        msg = "SeriesGroupBy.grouper is deprecated"
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            assert s.groupby(s).grouper.names == ["name"]
```

Comment:
pretty much all of these can just change grouper->_grouper i think, as long as there are a couple that test the deprecation